### PR TITLE
Throw exception after 30 min IDE DNS timeout.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

**Comment by nielsvm** [comment]
https://backlog.acquia.com/browse/DX-2732

File: src/Command/Ide/IdeCreateCommand.php [hunk 1]
```diff
@@ -96,7 +101,7 @@ protected function waitForDnsPropagation($ide_url): void {
         $this->logger->debug($e->getMessage());
       }
     });
-    LoopHelper::addTimeoutToLoop($loop, 15, $spinner, $this->output);
+    LoopHelper::addTimeoutToLoop($loop, 20, $spinner, $this->output);
```
Comment (nielsvm) [inline src/Command/Ide/IdeCreateCommand.php:104]: Why 20 minutes? I'm not aware of the average but wouldn't 15 suffice?

File: src/Helpers/LoopHelper.php [hunk 2]
```diff
@@ -47,8 +48,8 @@ public static function addTimeoutToLoop(
   ): void {
     $loop->addTimer($minutes * 60, function () use ($loop, $minutes, $spinner, $output) {
       self::finishSpinner($spinner);
-      $output->writeln("Timed out after $minutes minutes!");
       $loop->stop();
+      throw new AcquiaCliException("Timed out after $minutes minutes!");
```
Comment (nielsvm) [inline src/Helpers/LoopHelper.php:52]: This would skip printing the output URL right?

**Review by nielsvm — APPROVED** [review]

File: src/Command/Ide/IdeCreateCommand.php (hunk 1 above)
Comment (anavarre) [reply src/Command/Ide/IdeCreateCommand.php:104]: 15mn often fails for people not using Google or Cloudflare DNS servers. 20 is just buying us extra time, but won't fix the issue with DNS propagation times in Australia, India and parts of EMEA.

Comment (nielsvm) [reply src/Command/Ide/IdeCreateCommand.php:104]: That's good context to know, thanks.

File: src/Helpers/LoopHelper.php (hunk 2 above)
Comment (nielsvm) [reply src/Helpers/LoopHelper.php:52]: Confirmed my own question:
https://github.com/acquia/cli/pull/350/files/ff4887ad59953beac33f64417417273226f214f0#diff-dc099926d287b14cf26be796bb6b26f83b5c8541ff2b9f78dbb358871f1ecdf9R107

Comment (anavarre) [reply src/Helpers/LoopHelper.php:52]: I'm on the fence about doing this. Sometimes the timeout happens but URLs are actually accessible a few seconds/minutes after. I'd rather have a better message that explains it's not us, it's the DNS system, and the 2 URLs will shortly be available.

Comment (nielsvm) [reply src/Helpers/LoopHelper.php:52]: Its a failure nonetheless, and I think we should also consider removing the just created IDE if it expired DNS.

**Comment by danepowell** [comment]
Maybe also a message to the effect of "Acquia has been alerted to this issue, please follow status.acquia.com for updates."

**Comment by anavarre** [comment]
> Maybe also a message to the effect of "Acquia has been alerted to this issue, please follow status.acquia.com for updates."

Which wouldn't be true. You can have a DNS propagation issue on your end, or with the DNS system entirely. That wouldn't be an outage on our end nor would indicate status.acquia.com would necessarily be updated.

**Comment by nielsvm** [comment]
@danepowell yeah that's what I suggested in the filed ticket:
https://backlog.acquia.com/browse/DX-2732

**Comment by nielsvm** [comment]
@anavarre yeah so maybe we shouldn't be leveraging DNS checking from a users machine but from our own heart-beating logic?

**Comment by anavarre** [comment]
@nielsvm pre-provisioning should no longer have us run into the issue unless exceptional circumstances.

**Comment by nielsvm** [comment]
@grasmash how do you feel about relaying to `IdeDeleteCommand->execute()` as part of the exception handler for `CreateCommand`? I've got a strong feeling we shouldn't ever leave behind half-baked IDEs even if pre-provisioning will mask much of the issue here.

**Comment by grasmash** [comment]
I don’t feel like that’s the responsibility of the CLI tool. I agree that we should not leave around half baked IDEs, but that’s something that should be taken care of on the service side rather than the client side.

File: src/Helpers/LoopHelper.php (hunk 2 above)
Comment (grasmash) [reply src/Helpers/LoopHelper.php:52]: I'v changed this to catch the exception and print an error followed by the IDE links.

**Review by danepowell — APPROVED** [review]

**Pull request merged**